internal/ingress/nats/nats.go

```diff
@@ -0,0 +1,99 @@
+package nats
+
+import (
+	"encoding/json"
+
+	"github.com/kelindar/talaria/internal/config"
+	"github.com/kelindar/talaria/internal/monitor"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+	"github.com/nats-io/nats.go"
+)
+
+type Ingress struct {
+	// jetstream exposed interface.
+	jetstream JetstreamI
+	monitor   monitor.Monitor
+	conn      *nats.Conn
+}
+
+type jetstream struct {
+	// The name of the queue group.
+	queue   string // The name of subject listening to.
+	subject string
+	// The jetstream context which provide jetstream api.
+	jsContext nats.JetStreamContext
+}
+
+type JetstreamI interface {
+	// Subscribe to defined subject from Nats server.
+	Subscribe(handler nats.MsgHandler) (*nats.Subscription, error)
+	Publish(msg []byte) (nats.PubAckFuture, error)
+}
+
+type Event map[string]interface{}
+
+// New create new ingestion from nats jetstream to sinks.
+func New(conf *config.NATS, monitor monitor.Monitor) (*Ingress, error) {
+	nc, err := nats.Connect(conf.URL)
+	if err != nil {
+		return nil, err
+	}
+	js, err := nc.JetStream(nats.PublishAsyncMaxPending(256))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Ingress{
+		jetstream: NewJetStream(conf.Subject, conf.Queue, js),
+		monitor:   monitor,
+		conn:      nc,
+	}, nil
+}
+
+func NewJetStream(subject, queue string, js nats.JetStreamContext) *jetstream {
```

Comment:
add comments on functions specifying what they are supposed to do.

Reply:
Okayy will do.